File: backend/app/schemas/entity.py
from uuid import UUID

from pydantic import BaseModel


class EntityOut(BaseModel):
    id: UUID
    name: str
    entity_type: str
    description: str | None = None
    aliases: dict | None = None

    model_config = {"from_attributes": True}


class EntityRelationshipOut(BaseModel):
    id: UUID
    source_entity_id: UUID
    target_entity_id: UUID
    relationship_type: str
    weight: float
    details: dict | None = None

    model_config = {"from_attributes": True}


class EntityGraphOut(BaseModel):
    entities: list[EntityOut]
    relationships: list[EntityRelationshipOut]

File: frontend/src/types/entity.ts
export interface Entity {
  id: string;
  name: string;
  entity_type: "company" | "person" | "regulator" | "investor" | "sector";
  description: string | null;
  aliases: Record<string, string[]> | null;
}

export interface EntityRelationship {
  id: string;
  source_entity_id: string;
  target_entity_id: string;
  relationship_type: string;
  weight: number;
  details: Record<string, unknown> | null;
}

export interface EntityGraph {
  entities: Entity[];
  relationships: EntityRelationship[];
}
